## Assignment 3: Finish z-test lab
###### Author: Daniel Fay

In [1]:
# Grab time data from URL
!curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
107  1179  107  1179    0     0   8885      0 --:--:-- --:--:-- --:--:--  575k


In [1]:
# Import text file into pandas dataframe and print head
import pandas as pd
df = pd.read_csv('times.txt', sep=" ", header = None)
df.columns = ['times']
df.head()

times
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550


##### A new bus route (X8) has been implemented by the MTA. They want to know if it improves commute time?

Null Hypothesis (H0): The commute time is the same or longer with the implementation of the new bus route X8.  
(H0: U1 >= U2)

Alternative Hypthesis (Ha): The commute time is shorter with the implementation of the new bus route X8.  
(HA: U1 < U2)

Significance Level: alpha = 0.05  
Before Implementation: mean = 36, std =6

In [2]:
# Calculate new sample mean and Z score
import math
import scipy.stats as st
old_mean = 36
old_std = 6
new_mean = df['times'].mean()
z_score = (new_mean - old_mean)/(old_std/(math.sqrt(len(df))))
p_value = st.norm.sf(abs(z_score))
print ('Z Score = ',z_score)
print('P Value: ', p_value)

Z Score =  -2.5563971861666466
P Value:  0.00528811602207


#### Interpretation:
A left sided Z test of the hypothesis resulted in a Z Score of -2.556, which corresponds to a P Value of .005. Therefore since the p value is smaller than the stated significance level of .05 we can reject the null hypothesis and accept the alternative.

In [6]:
st.zscore(df.times)

array([-0.40046761, -0.23161084, -0.59664266, -0.42982385,  0.63991918,
        0.05011857,  0.76120804,  0.72009871, -0.48434457, -0.95806479,
       -2.19543256,  1.65672107, -0.62675957,  1.84950239, -2.52606981,
        1.12371567, -0.11840973,  1.10095887,  1.65303644, -0.5056658 ,
       -0.21377982, -1.16725225, -0.45189937,  0.12254923,  0.16539642,
        0.82472811,  0.50374176, -0.70522841, -1.48557234,  0.01378208,
        1.5435692 ,  0.63224858,  0.65448189,  0.73334125, -0.36255427,
       -1.66758595, -1.16553585,  1.27789555, -0.16974194, -0.39289393,
        0.37489254,  0.52866593, -0.01098638,  0.23205113,  1.35386622,
       -0.68534638,  0.36019916,  1.00536503,  1.39582312,  0.50673055,
       -1.6279505 ,  0.81752638, -0.40644479, -1.2090092 , -0.78502481,
        1.38994746,  0.32715959,  1.26660873,  0.87156302, -1.16874138,
       -1.31342794, -0.20423098, -0.22469338, -1.15195177,  0.70170609,
        0.25193599,  0.88051366,  0.8597691 , -2.05512768,  1.23